moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -537,7 +537,7 @@ bool constructConstraints(const rclcpp::Node::SharedPtr& node, const std::string
     return false;
 
   for (auto& constraint_id : constraint_ids)
-    constraint_id.insert(0, constraints_param);
+    constraint_id.insert(0, constraints_param + ".");
```

Comment:
If you add "." here, wouldn't it be redundant in line 488 etc?

Reply:
`constraint_param` is passed as parameter to the function, and its value is `constraints` (see [here](https://github.com/ros-planning/moveit2/blob/main/moveit_planners/ompl/ompl_interface/scripts/generate_state_database.cpp#L110) and [here](https://github.com/ros-planning/moveit2/blob/main/moveit_planners/ompl/ompl_interface/scripts/generate_state_database.cpp#L55))

---
The purpose of this for-loop is to prefix the `contraint_id` with the correct param "namespace". 
Assuming that before the for-loop, 
```cpp
constraint_ids = ["contraintA", "constraintB"];
```

After the loop, and **without the fix**, it becomes:
```cpp
contraint_ids =  ["contraintscontraintA", "contraintsconstraintB"];
```

Instead of (**with the fix**)
```cpp
contraint_ids =  ["contraints.contraintA", "contraints.constraintB"];
```
---
Line 488, also prefix (again) those `constraint_ids`, without the second fix, it becomes
```cpp
contraint_ids =  [".constraints.contraints.contraintA", ".constraints.contraints.constraintB"];
```
where the first `.` is not necessary, that's why I removed it.

--- 
I find the second prefixing (inside `collectConstraints`) confusing anyway. I guess it all boils down to how should the yaml file be structured. 
Do you guys have an example for this file ?

Reply:
To continue on my last remark, we could have the yaml look like this:
```yaml
generate_state_database:
  ros__parameters:
    planning_group: mygroup
    use_current_scene: false
    constraints:
      name: some_constraints
      constraint_ids: [constraintA]
      constraintA:
        type: joint
        frame_id: base_link
        joint_name: jr_link3_robot
        position: 0.0
        tolerance: 0.3
        weight: 1.0
```

In which case, the for-loop in the `constructConstraints` method would look like this:

```cpp
  for (auto& constraint_id : constraint_ids)
    constraint_id.insert(0, constraints_param + ".");
```

and inside the `collectConstraint` method we change line 479 to
```cpp
const auto constraint_param = constraint_id;
```
instead of ~~``const auto constraint_param = ".constraints." + constraint_id``~~